### PR TITLE
Use 'surface_azimuth' and 'surface_tilt' only, with pvlib convention

### PR DESCRIPTION
**Pull request by anomam**
Simplify the simulation API and fix potential inconsistencies in angle conventions between pvfactors and pvlib-python.
For array angles, the model should only rely on:
- ``surface_azimuth`` (with pvlib convention of 0 deg as North, and 90 as East)
- ``surface_tilt`` (with pvlib convention of values between 0 and 180 deg, so no more negative angles)

The previous inputs ``array_tilt`` and ``array_azimuth`` were confusing.

Addresses #8

### CODE REVIEW

File: pvfactors/docs/developer/pvfactors_demo.ipynb [hunk 1]
```diff
@@ -377,8 +409,8 @@
     "    'n_pvrows': 3,\n",
     "    'pvrow_height': 1.75,\n",
     "    'pvrow_width': 2.44,\n",
-    "    'array_azimuth': 90.,  # south-north oriented torque-tubes\n",
-    "    'array_tilt': 20.,  # positive rotation angles in the east\n",
+    "    'surface_azimuth': 180.,  # south-north oriented torque-tubes\n",
```
Comment (amir-asgharzadeh) [inline pvfactors/docs/developer/pvfactors_demo.ipynb]: As in above, comments need to be corrected here

File: pvfactors/pvarray.py [hunk 2]
```diff
@@ -133,10 +133,13 @@ class Array(ArrayBase):
...
-        have the same tilt angle [degrees]
-    :param float array_azimuth: azimuth angle of the whole array. All PV
-        rows must have the same azimuth angle [degrees]
+    :param float surface_tilt: Surface tilt angles in decimal degrees.
+        surface_tilt must be >=0 and <=180.
+        The tilt angle is defined as degrees from horizontal
+        (e.g. surface facing up = 0, surface facing horizon = 90)
+    :param float axis_azimuth: A value denoting the compass direction along
+        which the axis of rotation lies. Measured in decimal degrees East of
+        North.
```
Comment (amir-asgharzadeh) [inline pvfactors/pvarray.py]: Why define a new parameter? Can't we just use surface_azimuth here?

Comment (anomam) [reply pvfactors/pvarray.py]: Sorry I forgot to fix this docstring, thanks for catching this!

File: notebooks/pvfactors_demo.ipynb [hunk 3]
```diff
@@ -377,8 +409,8 @@
     "    'n_pvrows': 3,\n",
     "    'pvrow_height': 1.75,\n",
     "    'pvrow_width': 2.44,\n",
-    "    'array_azimuth': 90.,  # south-north oriented torque-tubes\n",
-    "    'array_tilt': 20.,  # positive rotation angles in the east\n",
+    "    'surface_azimuth': 180.,  # south-north oriented torque-tubes\n",
```
[the remark this answers is not in the document]
Comment (anomam) [reply notebooks/pvfactors_demo.ipynb]: Thanks for catching this!

**Comment by anomam** [comment]
Thanks a lot for the review @amir-asgharzadeh ! I fixed the issues you mentioned in 07269b3 .
I can now proceed to merging thanks to your review :+1:

**Pull request merged**